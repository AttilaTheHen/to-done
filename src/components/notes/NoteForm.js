import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NoteForm extends Component {
  state = {
    editing: false,
    key: null,
    completed: false,
    content: '',
    title: ''
  };

  static propTypes = {
    note: PropTypes.object
  };

  render() {
    const { completed, content, title } = this.state;

    return (
      <form>
        <InputControl name="title" value={title}/>
        <InputControl name="content" value={content}/>
        <input name="completed" value={completed} type="checkbox"/>
        <p>
          <button type="submit">Placeholder</button>
        </p>
      </form>
    );
  }
}

const InputControl = props => (
  <p>
    <label>
      {props.name}:
      <input {...props} required />
    </label>
  </p>
);
 
export default NoteForm;