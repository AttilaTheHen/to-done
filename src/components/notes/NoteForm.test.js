import NoteForm from './NoteForm';
import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Note form', () => {
  it('renders add if no note prop', () => {
    const handleComplete = jest.fn();
    const promise = Promise.resolve();
    handleComplete.mockReturnValueOnce(promise);

    const wrapper = mount(<NoteForm onComplete={handleComplete}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();

    const note = {
      title: 'Get veggies',
      content: 'Get some veggies',
      completed: false
    };

    wrapper.find('input[name="title"]').simulate('change', {
      target: {
        name: 'title',
        value: note.title
      }
    });

    wrapper.find('input[name="content"]').simulate('change', {
      target: {
        name: 'content',
        value: note.content
      }
    });

    wrapper.find('input[name="completed"]').simulate('change', {
      target: {
        name: 'completed',
        value: note.completed
      }
    });

    wrapper.find('button').simulate('submit');

    const calls = handleComplete.mock.calls;
    expect(calls.length).toBe(1);
    expect(calls[0][0]).toEqual(note);

    expect(toJSON(wrapper)).toMatchSnapshot();

    return promise
      .then(() => {
        wrapper.update();
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
  });
});