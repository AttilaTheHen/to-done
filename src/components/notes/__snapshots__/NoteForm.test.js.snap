// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note form renders add if no note prop 1`] = `
<NoteForm
  onComplete={[MockFunction]}
>
  <form>
    <InputControl
      name="title"
      value=""
    >
      <p>
        <label>
          title
          :
          <input
            name="title"
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="content"
      value=""
    >
      <p>
        <label>
          content
          :
          <input
            name="content"
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <input
      name="completed"
      type="checkbox"
      value={false}
    />
    <p>
      <button
        type="submit"
      >
        Placeholder
      </button>
    </p>
  </form>
</NoteForm>
`;

exports[`Note form renders add if no note prop 2`] = `
<NoteForm
  onComplete={[MockFunction]}
>
  <form>
    <InputControl
      name="title"
      value=""
    >
      <p>
        <label>
          title
          :
          <input
            name="title"
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="content"
      value=""
    >
      <p>
        <label>
          content
          :
          <input
            name="content"
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <input
      name="completed"
      type="checkbox"
      value={false}
    />
    <p>
      <button
        type="submit"
      >
        Placeholder
      </button>
    </p>
  </form>
</NoteForm>
`;
